lib/mayaUsd/ufe/UsdTRSUndoableCommandBase.cpp

```diff
@@ -0,0 +1,112 @@
+//
+// Copyright 2020 Autodesk
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+
+#include "UsdTRSUndoableCommandBase.h"
+#include "private/Utils.h"
+
+#include <ufe/scene.h>
+#include <ufe/sceneNotification.h>
+
+MAYAUSD_NS_DEF {
+namespace ufe {
+
+template<class V>
+UsdTRSUndoableCommandBase<V>::UsdTRSUndoableCommandBase(
+    const UsdSceneItem::Ptr& item, double x, double y, double z
+) : fItem(item), fNewValue(x, y, z)
+{}
+
+template<class V>
+void UsdTRSUndoableCommandBase<V>::initialize()
+{
+    if (cannotInit()) {
+        return;
+    }
+
+    // If prim does not have the attribute, add it.
+    if (!prim().HasAttribute(attributeName()))
+    {
+        fOpAdded = true;
+        addEmptyAttribute();
+    }
+
+    // See
+    // https://stackoverflow.com/questions/17853212/using-shared-from-this-in-templated-classes
+    // for explanation of this->shared_from_this() in templated class.
+    attribute().Get(&fPrevValue);
+    Ufe::Scene::instance().addObjectPathChangeObserver(this->shared_from_this());
+}
+
+template<class V>
+void UsdTRSUndoableCommandBase<V>::operator()(
```

Comment:
If the object we translate, rotate, scale sees its path changed (e.g. when reparenting or renaming), make sure we keep up with its new path.

Reply:
Let's keep this as a post review action, but I would prefer to unify how we are dealing with the problem (selection commands vs TRS command base) and rely on UFE path. Dealing with notifications can be tricky and surprising.